landlab/components/flow_accum/flow_accumulator.py

```diff
@@ -987,31 +984,40 @@ def accumulate_flow(self, update_flow_director=True):
             self._grid["node"]["flow__data_structure_delta"][:] = delta[1:]
 
             if self._is_raster:
-                tempD = BAD_INDEX_VALUE * np.ones((self._grid.number_of_links * 2))
+                tempD = BAD_INDEX_VALUE * np.ones(
+                    (self._grid.number_of_links * 2))
                 tempD[: len(D)] = D
-                self._grid["link"]["flow__data_structure_D"][:] = tempD.reshape(
-                    (self._grid.number_of_links, 2)
-                )
+                self._grid["link"]["flow__data_structure_D"][
+                    :] = tempD.reshape((self._grid.number_of_links, 2))
             else:
-                self._grid["link"]["flow__data_structure_D"][: len(D)] = D
+                self._grid["link"]["flow__data_structure_D"][:len(D)] = D
             self._grid["node"]["flow__upstream_node_order"][:] = s
 
-            # step 3. Run depression finder if passed
-            # at present this must go at the end.
+            # step 4. Accumulate (to one or to N depending on direction method)
+            a[:], q[:] = self._accumulate_A_Q_to_n(s, r, p)
 
-            # step 4. Accumulate (to one or to N depending on direction method. )
-            a, q = flow_accum_to_n.find_drainage_area_and_discharge_to_n(
-                s, r, p, self.node_cell_area, self._grid.at_node["water__unit_flux_in"]
-            )
-            # store drainage area and discharge.
-            self._grid["node"]["drainage_area"][:] = a
-            self._grid["node"]["surface_water__discharge"][:] = q
+        return (a, q)
```

Comment:
@SiccarPoint this revision of the flow accumulator run-one-step looks great!

Reply:
Sweet